utils/graceful_signals.go

```diff
@@ -16,16 +16,18 @@ type GracefulSignals struct {
 	handlers              []signalHandler
 	forceTerminateHandler func()
 	timeout               time.Duration
+	delay                 time.Duration
 	executed              bool
 
 	ch chan os.Signal
 	mu sync.Mutex
 }
 
 // Create new GracefulSignals struct.
-func NewGracefulSignals(timeout time.Duration) *GracefulSignals {
+func NewGracefulSignals(timeout time.Duration, delay time.Duration) *GracefulSignals {
```

Comment:
This feature is specific to the application, not to the GracefulSignals struct (which could be potentially extracted into a library).

We should also make the delay a part of the overall shutdown timeout (it's critical for platforms that have fixed or limited termination timeouts, like Heroku).

Thus, let's make this _sleep_ the first _shutdownable_ in the list of all shutdownables in `cli.go`.